string.c

```diff
@@ -0,0 +1,100 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <stdarg.h>
+#include <stdbool.h>
+#include <string.h>
+
+#include "common.h"
+#include "string.h"
+
+struct string_t *string_new(const char *data)
+{
+    if (data == NULL)
+        return NULL;
+    
+    struct string_t *s = malloc(sizeof(struct string_t));
+    s->data = (char*)(malloc(__CHAR_BIT__ * __STRING_DEFAULT_CAPACITY__));
+    s->capacity = __STRING_DEFAULT_CAPACITY__;
+    s->length = 0;
+
+    int i = 0;
+
+    while (*(data + i) != '\0') {
+        *(s->data + i) = *(data + i);
+        s->length++;
+        if (i++ == s->capacity) {
+            s->capacity *= 2;
```

Comment:
Bro why u change to *=2? Strings don't usually get that much bigger so it doesn't really make sense to have possibly 2x the amount of allocated memory compared to having a mear 32 bytes extra.